<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LogWindow</class>
 <widget class="QFrame" name="LogWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1128</width>
    <height>532</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>50</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QComboBox" name="logSeverityPicker">
         <property name="toolTip">
          <string>Select minimum message severity</string>
         </property>
         <property name="placeholderText">
          <string>Log Severity</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="SizeLineEdit" name="maxMessages">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>50</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Maximum messages</string>
         </property>
         <property name="text">
          <string>10000</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="topButton">
         <property name="toolTip">
          <string>Go to oldest message</string>
         </property>
         <property name="text">
          <string>Top</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bottomButton">
         <property name="toolTip">
          <string>Go to newest message</string>
         </property>
         <property name="text">
          <string>Bottom</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="clearButton">
         <property name="toolTip">
          <string>Remove all messages</string>
         </property>
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="logMessages">
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOn</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOn</enum>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="plainText">
        <string notr="true"/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SizeLineEdit</class>
   <extends>QLineEdit</extends>
   <header>gui/data.hxx</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>logSeverityPicker</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>LogWindow</receiver>
   <slot>setSeverity(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearButton</sender>
   <signal>clicked()</signal>
   <receiver>logMessages</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>441</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bottomButton</sender>
   <signal>clicked()</signal>
   <receiver>LogWindow</receiver>
   <slot>jumpLastMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>355</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topButton</sender>
   <signal>clicked()</signal>
   <receiver>LogWindow</receiver>
   <slot>jumpFirstMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxMessages</sender>
   <signal>editingFinished()</signal>
   <receiver>LogWindow</receiver>
   <slot>updateMaxMessages()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>265</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setSeverity(QString)</slot>
  <slot>jumpFirstMessage()</slot>
  <slot>jumpLastMessage()</slot>
  <slot>updateMaxMessages()</slot>
 </slots>
</ui>
